File: Data Structures & Algorithms/longest-consecutive-sequence/submission-12.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
       int longS = 1;
       int n = nums.size();
       if(n==0) return 0;
       sort(nums.begin(),nums.end());
       int curr = 1;

       //0,1,1,2,3,4,5,6

       for(int i = 1;i<n;i++){
        if(nums[i]!=nums[i-1]){
            if(nums[i]==nums[i-1]+1){
               curr++;
            }
             else {
                longS = max(longS,curr);
                curr = 1;
            }
        }
       

          
       }
       return max(longS,curr);


      
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-12.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        
        def backtrack(curr, start, sum_val):
            if sum_val == target:
                res.append(curr[:])
                return
            
            for i in range(start, len(nums)):
                if sum_val + nums[i] > target:
                    continue
                
                curr.append(nums[i])
                backtrack(curr, i, sum_val + nums[i])
                curr.pop()
        
        backtrack([], 0, 0)
        return res


